cache/models/samples.js

```diff
@@ -110,26 +112,33 @@ function parseName(name) {
  *  operation or not
  * @returns {Promise} - which resolves to true if the user has write permission
  */
-function checkWritePermission(aspect, userName, isBulk) {
-  let isWritable = true;
-  if (aspect.writers && aspect.writers.length) {
-    isWritable = aspect.writers.includes(userName);
-  }
+function checkWritePermission(aspectName, userName, isBulk) {
+  const redisCmds = [];
+  redisOps.checkAspectWriterCmds(redisCmds, aspectName, userName);
```

Comment:
In general, I find this "side-effects" style harder to read. I have to go into the "checkAspectWriterCmds" fn in order to see that it's pushing additional commands onto the array. IMO it's much clearer to have a fn called "getAspectWriterCmds" which just returns the array of those commands, rather than having the opaque side effect of array elements getting pushed onto one of the fn params.